What, behind the rabbit? It is the rabbit. (Added Killer Bunnies)

ChielBruin opened the pull request:
Added killer bunnies to the game.
The bunnies will hop around and walk towards the closest player or carrot.

Closes #265, closes #266 and closes #267.
Note: they do not yet kill the player as health is not implemented, but the code is there that should work when it is implemented.

ChielBruin committed:
Added KillerBunnies and Carrots

KillerBunnies have an AI that walks to the closest Carrot or player.

ChielBruin committed:
Removed test code

Removed some code used for testing so the build passes again.

ChielBruin committed:
Fixed a checkstyle warning

ChielBruin committed:
Minor changes

Added a setter and getter and added some javadocs.

ChielBruin committed:
Added tests for Carrot and KillerBunny

ChielBruin committed:
Created ai package

ChielBruin committed:
Added tests for the BunnyAI

Geweldig commented on src/main/java/nl/tudelft/contextproject/controller/GameController.java:
```diff
@@ -130,6 +130,13 @@ protected void attachLevel() {
+//		
+//		Entity c = new Carrot();
+//		c.move(4, 1, 4);
+//		game.getEntities().add(c);
```
Is this needed?

ChielBruin replied:
no it is not

ChielBruin committed:
Removed unneeded code

Geweldig commented on src/test/java/nl/tudelft/contextproject/model/entities/EnemyTest.java:
```diff
@@ -0,0 +1,64 @@
+	 * Getter for an {@link Enemy}.
+	 * 
+	 * @return
+	 * 		a fresh instance of {@link Enemy}.
```
. should be removed

Geweldig commented on src/main/java/nl/tudelft/contextproject/model/entities/Enemy.java:
```diff
@@ -0,0 +1,54 @@
+	 * Constructor that attaches the {@link EntityAI} to the entity.
+	 * 
+	 * @param ai
+	 * 		the {@link EntityAI} for this entity.
```
. should be removed

Geweldig commented on src/main/java/nl/tudelft/contextproject/model/entities/Enemy.java:
```diff
@@ -0,0 +1,54 @@
+	 * Set the ai of this {@link Enemy}.
+	 * 
+	 * @param ai
+	 * 		the new ai of this instance.
```
. should be removed

Geweldig commented on src/main/java/nl/tudelft/contextproject/model/entities/Enemy.java:
```diff
@@ -0,0 +1,54 @@
+	 * @param ai
+	 * 		the new ai of this instance.
+	 * @return
+	 * 		the old ai of this instance.
```
. should be removed

Geweldig commented on src/main/java/nl/tudelft/contextproject/model/entities/FoodItem.java:
```diff
@@ -0,0 +1,24 @@
+	 * Get the remaining amount.
+	 * 
+	 * @return
+	 * 		the remaining amount.
```
. should be removed

Geweldig commented on src/main/java/nl/tudelft/contextproject/model/entities/ai/EntityAI.java:
```diff
@@ -0,0 +1,26 @@
+	 * Set the owner of this ai.
+	 * 
+	 * @param owner
+	 * 		the owner of this ai.
```
. should be removed

Geweldig commented:
Other than my complaints I think this is good to merge!

ChielBruin committed:
Removed the FoodItem interface

ChielBruin committed:
Changed naming and improved VRPlayer

- Changed the name of Enemy to MovingEntity and EntityAI to EntityControl.
- Changed the vrPlayer to extend MovingEntity with a PlayerControl
- Fixed some tests that started failing

Diocruel commented:
Good to be merged!
